internal/ui/html/html.go

```diff
@@ -0,0 +1,217 @@
+func (v *View) ShowError(err error) error {
+	return v.basic.ShowError(err)
+}
+
+// ShowDiagnostics writes the given diagnostic to stdout in the formatted version.
+func (v *View) ShowDiagnostics(diags []*pb.Diagnostic) error {
+	return v.basic.ShowDiagnostics(diags)
+}
+
+// ShowVersion shows the version information.
+func (v *View) ShowVersion(all bool, res *pb.GetVersionResponse) error {
+	return v.ShowError(status.Unimplemented("html/ui: ShowVersion"))
+}
+
+// ShowScenariosValidateConfig shows the validation response.
+func (v *View) ShowScenariosValidateConfig(res *pb.ValidateScenariosConfigurationResponse) error {
+	return v.ShowError(status.Unimplemented("html/ui: ShowScenariosValidateConfig"))
+}
+
+// ShowScenarioList shows the a list of scenarios.
+func (v *View) ShowScenarioList(res *pb.ListScenariosResponse) error {
+	return v.ShowError(status.Unimplemented("html/ui: ShowScenarioList"))
+}
+
+// ShowScenarioOutline shows the scenario outlines.
+func (v *View) ShowScenarioOutline(res *pb.OutlineScenariosResponse) error {
+	if res == nil || res.GetOutlines() == nil || len(res.GetOutlines()) < 1 {
+		return nil
+	}
+
+	t, err := template.New("outline.html.tmpl").Funcs(template.FuncMap{
+		"scenarioName": func(g *pb.Scenario_Outline) string {
+			if g == nil {
+				return ""
+			}
+
+			return g.GetScenario().GetId().GetName()
+		},
+		"scenarioDescription": func(g *pb.Scenario_Outline) string {
+			if g == nil {
+				return ""
+			}
+
+			return g.GetScenario().GetId().GetDescription()
+		},
+		"scenarioVariants": func(g *pb.Scenario_Outline) []*pb.Matrix_Vector {
+			if g == nil {
+				return nil
+			}
+
+			return g.GetMatrix().GetVectors()
+		},
+		"scenarioSteps": func(g *pb.Scenario_Outline) []*pb.Scenario_Outline_Step {
+			if g == nil {
+				return nil
+			}
+
+			return g.GetSteps()
+		},
+		"scenarioVerifies": func(g *pb.Scenario_Outline) []*pb.Quality {
+			if g == nil {
+				return nil
+			}
+
+			return g.GetVerifies()
+		},
+		"scenarioStepVerifies": func(g *pb.Scenario_Outline_Step) []*pb.Quality {
+			if g == nil {
+				return nil
+			}
+
+			return g.GetVerifies()
+		},
+		"printVariantKey": func(vec *pb.Matrix_Vector) string {
+			if vec == nil || vec.GetElements() == nil || len(vec.GetElements()) < 1 {
+				return ""
+			}
+
+			return vec.GetElements()[0].GetKey()
+		},
+		"printVariantValues": func(vec *pb.Matrix_Vector) string {
+			if vec == nil || vec.GetElements() == nil || len(vec.GetElements()) < 1 {
+				return ""
+			}
+			b := new(strings.Builder)
+			for i := range vec.GetElements() {
+				if i != 0 {
+					fmt.Fprint(b, ", ")
+				}
+				fmt.Fprintf(b, "%s", vec.GetElements()[i].GetValue())
+			}
+
+			return b.String()
+		},
+	}).ParseFS(templates, "template/outline.html.tmpl")
+	if err != nil {
+		return v.ShowError(err)
+	}
+
+	buf := bytes.Buffer{}
+	err = t.Execute(&buf, res)
+	if err != nil {
+		return v.ShowError(err)
+	}
+
+	v.basic.UI().Output(buf.String())
```

Comment:
I doubt this will happen but `UI()` can return a nil pointer and this could panic.

Reply:
We have nil guards on the access so it should be safe. I added a test case to check anyway.